packages/@aws-cdk/pipelines/lib/main/pipeline.ts

```diff
@@ -112,10 +109,15 @@ export class Pipeline extends CoreConstruct {
     if (this.built) {
       throw new Error('build() has already been called: can only call it once');
     }
-    this.engine.buildDeployment({ blueprint: this.blueprint, scope: this });
+    this.doBuildPipeline();
     this.built = true;
   }
 
+  /**
+   * Implemented by subclasses to do the actual pipeline construction
+   */
+  protected abstract doBuildPipeline(): void;
```

Comment:
Maybe:

```suggestion
  protected abstract constructPipeline(): void;
```

Reply:
It's the "actual" implementation of `buildPipeline()`, so I feel `doBuild...` makes sense here. I understand it offends your sensibilities, it does mine too a little 😅... but it's a pretty common pattern to have a public `foo()` method for some checks, and then a `doFoo()` method for the "actual work".

Reply:
OK.